File: 1.09/erlang/second.erl
-module(second).
-export([hypotenuse/2,perimeter_right_triangle/2,area_right_triangle/2]).


% Using the function square from first.erl, define a function that gives the
% size of the hypotenuse of a right-angled triangle given the lengths of the
% two other sides.

hypotenuse(A,B) ->
  math:sqrt(A*A + B*B).

% Define functions that give the perimeter and area of a right-angled triangle,
% given the lengths of the two short sides.

perimeter_right_triangle(A,B) ->
  (A + B)/2.

area_right_triangle(A,B) ->
  (A*A + B*B)/2.

File: 1.09/elixir/second.ex
# Using the function square from first.erl, define a function that gives the
# size of the hypotenuse of a right-angled triangle given the lengths of the
# two other sides.

defmodule Second do

  # Using the function square from first.erl, define a function that gives the
  # size of the hypotenuse of a right-angled triangle given the lengths of the
  # two other sides.

  def hypotenuse(a,b), do: :math.sqrt(a*a + b*b)

  # Define functions that give the perimeter and area of a right-angled triangle,
  # given the lengths of the two short sides.

  def perimeter_right_triangle(a, b), do: (a + b)/2

  def area_right_triangle(a, b), do: (a*a + b*b)/2
end
